airbyte-integrations/connectors/source-amazon-ads/unit_tests/test_streams.py

```diff
@@ -0,0 +1,306 @@
+        response_body = campaigns[start_index : start_index + count]
+        return (200, {}, json.dumps(response_body))
+
+    responses.add_callback(
+        responses.GET,
+        "https://advertising-api.amazon.com/sd/campaigns",
+        content_type="application/json",
+        callback=campaigns_paginated_response_cb,
+    )
+    profile_records = get_all_stream_records(profile_stream)
+
+    campaigns_records = get_all_stream_records(campaigns_stream)
+    assert len(campaigns_records) == len(profile_records) * len(json.loads(campaigns_response))
+
+
+@pytest.mark.parametrize(("status_code"), [HTTPStatus.FORBIDDEN, HTTPStatus.UNAUTHORIZED])
+@responses.activate
+def test_streams_campaigns_pagination_403_error(mocker, status_code, test_config, profiles_response, campaigns_response):
+    setup_responses(
+        profiles_response=profiles_response,
+    )
+    responses.add(
+        responses.GET,
+        "https://advertising-api.amazon.com/sd/campaigns",
+        json={"message": "msg"},
+        status=status_code,
+    )
+    source = SourceAmazonAds()
+    streams = source.streams(test_config)
+    campaigns_stream = get_stream_by_name(streams, "sponsored_display_campaigns")
+
+    with pytest.raises(requests.exceptions.HTTPError):
+        get_all_stream_records(campaigns_stream)
+
+
+@responses.activate
+def test_streams_campaigns_pagination_403_error_expected(mocker, test_config, profiles_response, campaigns_response):
+    setup_responses(
+        profiles_response=profiles_response,
+    )
+    responses.add(
+        responses.GET,
+        "https://advertising-api.amazon.com/sd/campaigns",
+        json={"code": "403", "details": "details", "requestId": "xxx"},
+        status=403,
+    )
+    source = SourceAmazonAds()
+    streams = source.streams(test_config)
+    campaigns_stream = get_stream_by_name(streams, "sponsored_display_campaigns")
+
+    campaigns_records = get_all_stream_records(campaigns_stream)
+    assert campaigns_records == []
+
+
+@pytest.mark.parametrize(
+    ("stream_name", "endpoint"),
+    [
+        ("sponsored_display_ad_groups", "sd/adGroups"),
+        ("sponsored_display_product_ads", "sd/productAds"),
+        ("sponsored_display_targetings", "sd/targets"),
+    ],
+)
+@responses.activate
+def test_streams_displays(
+    test_config,
+    stream_name,
+    endpoint,
+    profiles_response,
+    adgroups_response,
+    targeting_response,
+    product_ads_response,
+):
+    setup_responses(
+        profiles_response=profiles_response,
+        adgroups_response=adgroups_response,
+        targeting_response=targeting_response,
+        product_ads_response=product_ads_response,
+    )
+
+    source = SourceAmazonAds()
+    streams = source.streams(test_config)
+    profile_stream = get_stream_by_name(streams, "profiles")
+    test_stream = get_stream_by_name(streams, stream_name)
+    _ = get_all_stream_records(profile_stream)
```

Comment:
Please add a comment which will explain why you just calling `get_all_stream_records()` here without saving the result and using it later.

Reply:
In fact we dont need it, this line was added when prefetching profiles record was not implemented. Removed it